x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/Fips140ProviderVerificationTests.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.core;
+
+import org.bouncycastle.crypto.CryptoServicesRegistrar;
+import org.elasticsearch.test.ESTestCase;
+
+import java.security.Security;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+
+public class Fips140ProviderVerificationTests extends ESTestCase {
+
+    public void testBcFipsProviderInUse(){
+        if (inFipsJvm()){
+            assertThat(Security.getProviders().length > 0, equalTo(true));
+            assertThat(Security.getProviders()[0].getName(), containsString("BCFIPS"));
+        }
+    }
+
+    public void testInApprovedOnlyMode(){
+        if (inFipsJvm()){
+            assertThat(CryptoServicesRegistrar.isInApprovedOnlyMode(), equalTo(true));
+        }
+    }
```

Comment:
I think these tests can have better coverage if we move them into `ESTestCase#inFipsJvm` so that they are always checked if fips is enabled.

Reply:
I think it's better here. I feel like this is more explicit in its own test that has its own distinct purpose rather than burying it in another test (case).

We might get more coverage by moving them, but I don't understand the value of that - we only need to run them once to get the value we need.

Reply:
I think it has some value, maybe not for CI, but for individual tests when you need run them locally so that you can be sure FIPS is always correctly configured by the property passed from CLI.

Reply:
> I think it has some value, maybe not for CI, 

The purpose is to cover CI ,do that we don't make any change in the feature that would lead to us not running CI in non approved mode. 

I think the clarity of it being in it's own class makes sense and we should leave it here